### PR TITLE
ug: Add user guides on custom CI for OS and apps

### PR DESCRIPTION
**Pull request by mike-sul**
Signed-off-by: Mike <mike.sul@foundries.io>

## Readiness

* [x] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

_Why merge this PR? What does it solve?_

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [x] Run spelling and grammar check, preferably with linter.
* [x] Avoid changing any header associated with a link/reference.
* [x] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [x] Match tone and style of page/section.
* [x] Run `make linkcheck`.
* [x] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [x] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [x] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 1ebed4f416ac84d2d325454fc3133a10c7f3c8a3 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1849/docs/artifacts/html/index.html

**Comment by mike-sul** [comment]
I merged two PRs with two guides  into a single PR and introduced an entry page referenced from the main TOC.

**Comment by doanac** [comment]
Docs for ae080e838174a0c6ff6b18cb02b1157665bbfd5e are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1850/docs/artifacts/html/index.html

**Comment by mike-sul** [comment]
@kprosise The ``vale`` complains only for the `linux` in references.

```
vale source/user-guide/custom-ci/custom-ci* --minAlertLevel=error

 source/user-guide/custom-ci/custom-ci-for-rootfs.rst
 8:80   error  Use 'Linux' instead of 'linux'  Fio-docs.Branding-and-names 
 21:15  error  Use 'LmP' instead of 'lmp'      Fio-docs.Branding-and-names 
 21:37  error  Use 'Linux' instead of 'linux'  Fio-docs.Branding-and-names 
 51:15  error  Use 'Linux' instead of 'linux'  Fio-docs.Branding-and-names 

```

File: source/user-guide/custom-ci/custom-ci-for-apps.rst [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+your App source code in the FoundriesFactory git repository and using the FoundriesFactory CI service.
+Therefore, you can host your App in any source code repository and build App by any other framework,
+yet still leverage the rest part of FoundriesFactory.
+
+This section guides you through the steps of creating a custom CI pipeline in GitHub that:
+
+- builds multi-arch container images and pushes them to `Foundries Registry`_;
+- builds `Foundries Compose App`_ and pushes it `Foundries Registry`_;
+- composes `TUF Targets role metadata`_ compliant with FoundriesFactory TUF requirements;
+- adds the composed TUF Targets to `FoundriesFactory Targets`_.
```
Comment (doanac) [inline source/user-guide/custom-ci/custom-ci-for-apps.rst:24]: i'm still finding the mix of semi-colons and a period odd. @kprosise should have an opinion if she hasn't already

File: source/user-guide/custom-ci/custom-ci-for-rootfs.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
+.. _ug-custom-ci-for-rootfs:
+
+Custom CI For RootFS
```
Comment (doanac) [inline source/user-guide/custom-ci/custom-ci-for-rootfs.rst:3]: We might want to think about this "RootFS" term. I'm not sure we've used this term before. I can see us explaining this to people a lot by saying something like "You know, a Platform build". So I wonder if we should call it something like that Yocto/OE/Platform instead.

**Review by doanac — COMMENTED** [review]
I like this shape of this new PR.

File: source/user-guide/custom-ci/custom-ci-for-apps.rst (hunk 1 above)
Comment (mike-sul) [reply source/user-guide/custom-ci/custom-ci-for-apps.rst:24]: That's not exactly mix, this is one sentence where items are listed after `:`, separated by `;` and then sentence ends with `.`. At least, this is how I was taught.

**Comment by doanac** [comment]
Docs for 262ae9ab391fb963d4be4c2733ebad363eeaa63e are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1863/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 39055fc67787085294153c4525d4e150a0907788 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1866/docs/artifacts/html/index.html

**Comment by kprosise** [comment]
Hi @mike-sul! I will be reviewing this tomorrow, I have not forgotten!

File: source/user-guide/custom-ci/custom-ci-for-apps.rst (hunk 1 above)
Comment (kprosise) [reply source/user-guide/custom-ci/custom-ci-for-apps.rst:24]: This is correct, and my opinion is to keep it.
That said, above any "rule" comes readability.
Once in a great while content can follow all the rules, but may be difficult to parse, or the formatting otherwise makes it difficult to read.
There can also be content that ignores some rules but the result is clear and easy to grok.
But breaking rules makes things inconsistent, which is a whole different problem, so we should really try and follow the rules.

TL;DR: this is fine, as is content that knows when to break the rules; readability is paramount

File: source/user-guide/custom-ci/custom-ci-for-rootfs.rst (hunk 2 above)
Comment (kprosise) [reply source/user-guide/custom-ci/custom-ci-for-rootfs.rst:3]: I was able to assume the correct context by the title. but that's likely my familiarity.
My recommendation would be to update `custom-ci.rst` to have a description of each child page, and what it will be discussing similar to the [FoundriesFactory](https://docs.foundries.io/latest/reference-manual/factory/factory.html) section.
This is actually something I want to implement throughout our docs. However it is likely best to handle that separately.

**Comment by doanac** [comment]
Docs for 62329fb2f87c1a92382657455732d2584079be08 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1875/docs/artifacts/html/index.html

File: source/user-guide/custom-ci/custom-ci-for-rootfs.rst (hunk 2 above)
Comment (mike-sul) [reply source/user-guide/custom-ci/custom-ci-for-rootfs.rst:3]: >My recommendation would be to update custom-ci.rst to have a description of each child page, and what it will be discussing similar to the [FoundriesFactory](https://docs.foundries.io/latest/reference-manual/factory/factory.html) section.

I updated `custom-ci` with "what it will be discussing".

File: source/user-guide/custom-ci/custom-ci-for-rootfs.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+Push OSTree Repo To Cloud
+-------------------------
+The :ref:`ref-linux-dev-container` includes utilities called fiopush and fiocheck that pushes an ostree repo to the foundries multi-tenant storage based on GCS.
+You need an auth token to run these commands.
+The token can be obtained at `FoundriesFactory WebApp`_, and it should have ``targets:read-update`` scope.
+Run ``fiopush -factory <factory> -repo ./deploy/images/intel-corei7-64/ostree_repo -token <fio-token>`` to push the bitbaked ostree repo to the FoundriesFactory storage.
+
+
+Add OSTree Target
+-----------------
```
Comment (kprosise) [inline source/user-guide/custom-ci/custom-ci-for-rootfs.rst:58]:
```suggestion
-----------------

```

File: source/user-guide/custom-ci/custom-ci-for-rootfs.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+Once the ostree repo carrying rootfs is built and pushed to the cloud, a user can add a new Target referring to it.
+
+The rootfs committed to the ostree repo is referred by the commit hash.
+Run ``find ./deploy/images/intel-corei7-64/ostree_repo -name *.commit`` or ``ostree --repo ./deploy/images/intel-corei7-64/ostree_repo rev-parse <machine|hardware ID>``
+to obtain the hash.
+
+Then, you can run the ``fioctl targets add`` add new Target that refer to the given ostree-based rootfs, e.g.
```
Comment (kprosise) [inline source/user-guide/custom-ci/custom-ci-for-rootfs.rst:65]:
```suggestion
Then, you can run ``fioctl targets add`` to add a new Target that references the given ostree-based rootfs, e.g.,
```

File: source/user-guide/custom-ci/custom-ci-for-rootfs.rst [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+.. prompt:: text
+
+    fioctl targets add --type ostree --tags master,devel --src-tag master intel-corei7-64 094a6d77b7053f2fec1e5e4ccd83c38cb89174f644303c6bb09693648be98912
+
+Check Created OSTree Target
```
Comment (kprosise) [inline source/user-guide/custom-ci/custom-ci-for-rootfs.rst:71]:
```suggestion
Check the OSTree Target
```

File: source/user-guide/custom-ci/custom-ci-for-rootfs.rst [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+---------------------------
```
Comment (kprosise) [inline source/user-guide/custom-ci/custom-ci-for-rootfs.rst:72]:
```suggestion
-----------------------

```

File: source/user-guide/custom-ci/custom-ci-for-rootfs.rst [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+Use ``fioctl targets list`` and ``fioctl targets show`` commands to check whether the new Target is registered in the Foundries OTA service
```
Comment (kprosise) [inline source/user-guide/custom-ci/custom-ci-for-rootfs.rst:73]:
```suggestion
Use ``fioctl targets list`` and ``fioctl targets show`` commands to check whether the new Target is registered in the OTA service
```

File: source/user-guide/custom-ci/custom-ci-for-rootfs.rst [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+Then during the update one can see the log saying that aklite is downloading the expected ostree commit.
+
+.. prompt:: text
+
+    info: Fetching ostree commit 094a6d77b7053f2fec1e5e4ccd83c38cb89174f644303c6bb09693648be98912 from https://storage.googleapis.com/ota-lite-ostree-eu/094a6d77b7053f2fec1e5e4ccd83c38cb89174f644303c6bb09693648be98912
+    ...
+    aktualizr-lite status
+    info: Active image is: 1589	sha256:00b2ad4a1dd7fe1e856a6d607ed492c354a423be22a44bad644092bb275e12fa
+
```
Comment (kprosise) [inline source/user-guide/custom-ci/custom-ci-for-rootfs.rst:92]:
```suggestion
```

**Review by kprosise — APPROVED** [review]
Finally finished the review, lots of suggestions, however they're mostly small things

**Comment by mike-sul** [comment]
@kprosise Updated according to your suggestions.

**Comment by doanac** [comment]
Docs for fc62e7d2d0660d19d7c90a2c0a8cf1a0adf0ccab are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1876/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for b5b302907cc22910d17e81baa4f92fa8db812674 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1877/docs/artifacts/html/index.html
